MetadataExtractor/IO/RandomAccessStream.cs

```diff
@@ -414,16 +429,18 @@ public void Seek(long index, bool isSequential = false)
         /// Ensures that the buffered bytes extend to cover the specified index. If not, an attempt is made
         /// to read to that point.
         /// </summary>
+        /// <returns>The number of bytes available out of the number of bytes requested</returns>
         /// <remarks>
         /// If the stream ends before the point is reached, a <see cref="BufferBoundsException"/> is raised.
+        /// Requesting more bytes than available raises an exception if <paramref name="allowPartial"/> is false
         /// </remarks>
         /// <param name="index">the index from which the required bytes start</param>
         /// <param name="bytesRequested">the number of bytes which are required</param>
         /// <param name="isSequential">flag indicating if caller is using sequential access</param>
         /// <param name="allowPartial">flag indicating whether count should be enforced when validating the index</param>
         /// <exception cref="BufferBoundsException">negative index, less than 0 bytes, or too many bytes are requested</exception>
         /// <exception cref="IOException">if the stream ends before the required number of bytes are acquired</exception>
-        public long ValidateIndex(long index, long bytesRequested, bool isSequential, bool allowPartial = false)
+        internal long ValidateIndex(long index, long bytesRequested, bool isSequential, bool allowPartial = false)
```

Comment:
Perhaps name this `ValidateRange`.

Also it looks like `allowPartial` is always known at the call site. Perhaps break this in two (adding `ValidateRangeAllowPartial` or similar) so there's no perf hit when `allowPartial` is `true`.

Reply:
Renamed, and removed allowPartial entirely